content/beginner/194_secrets_manager/deploy-workload-secret.md

```diff
@@ -0,0 +1,114 @@
+---
+title: "Deploy pods with mounted secrets"
+date: 2021-10-01T00:00:00-04:00
+weight: 30
+pre: '<i class="fa fa-film" aria-hidden="true"></i>'
+draft: false
+---
+
+#### Create SecretProviderClass
+Create SecretProviderClass custom resource with ```provider:aws``` . The SecretProviderClass must be in the same namespace as the pod using it later.
+
+```bash
+cat << EOF > nginx-deployment-spc.yaml
+---
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: nginx-deployment-spc
+spec:
+  provider: aws
+  parameters:
+    objects: |
+        - objectName: "DBSecret_eksworkshop"
+          objectType: "secretsmanager"
+EOF
+```
+
+Create custom resource.
+```bash
+kubectl apply -f nginx-deployment-spc.yaml
+
+kubectl get SecretProviderClass
+```
+
+The output indicates the resource created successfully.
+{{<output>}}
+NAME                   AGE
+nginx-deployment-spc   18s
+{{</output>}}
+
+
+
+#### Create pod and mount secrets
+
+Configure a pod to mount volumes based on the SecretProviderClass ```nginx-deployment-spc``` created earlier and to retrieve secrets from the AWS Secrets Manager. The pod is also using a service account ```nginx-deployment-sa``` bound to the IAM role. 
+
+```bash
+cat << EOF > nginx-deployment.yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      serviceAccountName: nginx-deployment-sa
+      containers:
+      - name: nginx-deployment
+        image: nginx
+        ports:
+        - containerPort: 80
+        volumeMounts:
+        - name: secrets-store-inline
+          mountPath: "/mnt/secrets"
+          readOnly: true
+      volumes:
+      - name: secrets-store-inline
+        csi:
+          driver: secrets-store.csi.k8s.io
+          readOnly: true
+          volumeAttributes:
+            secretProviderClass: nginx-deployment-spc
+EOF
+```
+
+Create the deployment and verify the creation of pods.
+```bash
+kubectl apply -f nginx-deployment.yaml
+sleep 5
+kubectl get pods -l "app=nginx"
+```
+{{<output>}}
+NAME                                READY   STATUS    RESTARTS   AGE
+nginx-deployment-75bfbbcf99-ws4ts   1/1     Running   0          10s
+{{</output>}}
+
+
+#### Verify the mounted secret
+
+Finally, verify the secret mounted as a file by executing the command within the pod.
+
+```bash
+kubectl exec -it $(kubectl get pods | awk '/nginx-deployment/{print $1}' | head -1) \
```

Comment:
Change this to multiline where the pod name is declared on one line as an env and the second uses that name (for convention as well as readability to the end user).
```
export POD_NAME=$(kubectl get pods -l app=nginx -o jsonpath='{.items[].metadata.name}')
kubectl exec -it ${POD_NAME} -- cat /mnt/secrets/DBSecret_eksworkshop; echo
```

Reply:
change kubectl exec command. Get ${POD_NAME} in a variable separately first, as suggested.

Changes applied with https://github.com/aws-samples/eks-workshop/pull/1271/commits/3b2252ac93019d6e453fef8e003a572ad629c472